src/pages/index.tsx

```diff
@@ -4,19 +4,108 @@ import { Layout } from "../components/Layout";
 import { useRouter } from "next/router";
 import dynamic from "next/dynamic";
 import { Posts } from "../components/Post";
+import { api } from "../utils/api";
+import { useEffect, useState } from "react";
+import { BlogPostCategory } from "../types/blogPost";
+import SearchPost from "../components/SearchPost";
+import { usePagination } from "../hooks/usePagination";
+import { useDebounce } from "../hooks/useDebounce";
+import { Pagination } from "../components/Pagination";
 
 const HomePage: NextPage = () => {
+  const [paginationPage, setPaginationPage] = useState(0);
+  const [searchActive, SetSearchActive] = useState(false);
+  const [searchCategory, setSearchCategory] = useState<BlogPostCategory[]>([]);
+  const [searchWord, setSearchWord] = useState<string>("");
+  const debounce = useDebounce(searchWord, 500);
   const router = useRouter();
+
+  useEffect(() => {
```

Comment:
It doesn't work - I type some text, refresh the page, and text is not inside text input

Please watch and understand this https://www.youtube.com/watch?v=L6BE-U3oy80. This is about zod, a the typescript validation library.

Then you can create a query params schema
```
const queryParamsSchema = z.object({
  search: z.string().optional().default(''),
  categories: z.array(z.nativeEnum(BlogPostCategory)).optional().default([]),
});
```
(Categories won't work, because there will be a string with comma separated categories instead of array. Please do a research how to solve it)

Then, you can simply
```
const router = useRouter();
const [postListFilters, setPostListFilters] = useState(queryParamsSchema.parse(router.query));
```